### PR TITLE
Improve Code Quality 3

### PR DESCRIPTION
**Pull request by iamcarbon**
This PR:

- Compiles regular expressions
- Adds a timeout to regular expressions that may be from an untrusted source
- Replaces a few method calls with constant pattern matching
- Marks additional fields as readonly by removing private setters

## Prerequisites

Please make sure you can check the following two boxes:

- [x] I have read the **CONTRIBUTING** document
- [x] My code follows the code style of this project

## Contribution Type

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [x] All new and existing tests passed

### CODE REVIEW

File: src/AngleSharp/Html/InputTypes/ColorInputType.cs [hunk 1]
```diff
@@ -8,7 +8,7 @@ class ColorInputType : BaseInputType
     {
         #region Fields
 
-        static readonly Regex color = new Regex("^\\#[0-9A-Fa-f]{6}$");
+        static readonly Regex hexColorPattern = new Regex("^\\#[0-9A-Fa-f]{6}$", RegexOptions.IgnoreCase | RegexOptions.Compiled);
```
Comment (FlorianRappl) [inline src/AngleSharp/Html/InputTypes/ColorInputType.cs]: Either ignore case and only have 0-9A-F or don't ignore case and keep the pattern. But having both is a bit inefficient.

**Review by FlorianRappl — COMMENTED** [review]
Another great PR. :beers:

File: src/AngleSharp/Html/InputTypes/ColorInputType.cs (hunk 1 above)
Comment (iamcarbon) [reply src/AngleSharp/Html/InputTypes/ColorInputType.cs]: Good catch. Updated.

**Comment by iamcarbon** [comment]
Ready for review.

**Review by FlorianRappl — APPROVED** [review]
LGTM - thanks! :beers:

**Pull request merged**